zcash_transparent/src/bundle.rs

```diff
@@ -258,6 +266,29 @@ impl TxIn<Authorized> {
     }
 }
 
+impl TxIn<builder::Coinbase> {
+    /// Creates an input for a coinbase transaction. Does not support creating inputs for the
+    /// genesis block.
+    pub fn coinbase(
+        height: BlockHeight,
+        miner_data: &MinerData,
+        sequence: u32,
+    ) -> Result<Self, coinbase::Error> {
+        let height = i64::from(height);
+
+        // Serialize the height. The serialization is specified in
+        // <https://zips.z.cash/protocol/protocol.pdf#txnconsensus>.
+        let mut script_sig = match height {
+            0 => Err(coinbase::Error::GenesisInputNotSupported)?,
+            _ => zcash_script::pattern::push_num(height).to_bytes(),
+        };
+
+        script_sig.extend(miner_data.as_ref());
```

Comment:
It's been long enough since I messed with coinbase inputs that I've forgotten: is this correct for how miner data is appended? Should it not be in a `PushData`? Document the answer in this code so it's clear to the next reviewer.

Reply:
I'm looking at how `zcashd` does it.

If I understand the code correctly, it doesn't use miner data at all in the `getblocktemplate` RPC: https://github.com/zcash/zcash/blob/16ac743764a513e41dafb2cd79c2417c5bb41e81/src/miner.cpp#L308.

It does use optional miner data in its internal miner, though: https://github.com/zcash/zcash/blob/16ac743764a513e41dafb2cd79c2417c5bb41e81/src/miner.cpp#L802; and serializes it as a script. I'll do that, then.